pkgdb/src/realisepkgs/realise.cc

```diff
@@ -385,6 +390,11 @@ getRealisedOutputs( nix::ref<nix::EvalState> &       state,
   // we need to build the derivation to get all outputs
   if ( ! allValid )
     {
+      debugLog(
+        nix::fmt( "getting cursor for %s", lockedPackage.attrPath[0] ) );
+      auto cursor  = evalCacheCursorForInput( state,
+                                             lockedPackage.input,
+                                             lockedPackage.attrPath );
```

Comment:
note nonblocking: it would probably make more sense to refactor all of this to have the fast path where we don't need an eval cursor and a slow path where we do, rather than interleaving the two paths and having to get the cursor twice

Reply:
I agree.  My main purpose was a quick fix for now.  I think ideally we move the quick check into Rust and never get here unless we really need it.  This was _safest_ for now IMO.